### PR TITLE
Add GeoTrellisRasterSource compat with GT 1.x and 2.x layers

### LINKED ISSUE #3179: GeoTrellisRasterSources legacy and temporal layers support
At this point GeoTrellisRasterSource can not read legacy (1.x and 2.x) GeoTrellis layers due to the `Key` class name change (from `geotrellis.spark.SpatialKey` to `geotrellis.layer.SpatialKey`). 

The task is to add a compact / fallback logic to read old layers. 
This is importat to make GeoTrellis-server OGC examples work.

### PR DESCRIPTION
**Pull request by pomadchin**
# Overview

This PR makes possible GeoTrellisRasterSources to work with old GeoTrellis layers

## Checklist

- [x] [docs/CHANGELOG.rst](https://github.com/locationtech/geotrellis/blob/master/CHANGELOG.md) updated, if necessary

## Demo

https://github.com/geotrellis/geotrellis-server/pull/191

Closes #3179

### CODE REVIEW

File: project/Settings.scala [hunk 1]
```diff
@@ -109,6 +111,21 @@ object Settings {
...
+        def findYear(header: String): Option[String] = header match {
+          case Pattern(years, _) => Some(years)
+          case _                 => None
+        }
+        def apply(text: String, existingText: Option[String]): String = {
+          // preserve year of old headers
+          val newText = CommentStyle.cStyleBlockComment.commentCreator.apply(text, existingText)
+          existingText.flatMap(_ => existingText.map(_.trim)).getOrElse(newText)
+        }
+      })
```
Comment (echeipesh) [inline project/Settings.scala]: Did moving the config here fix some issue with header generation?

File: store/src/main/scala/geotrellis/store/GeoTrellisRasterSource.scala [hunk 2]
```diff
@@ -174,25 +174,43 @@ object GeoTrellisRasterSource {
...
+      (Class.forName(header.keyClass), Class.forName(header.valueClass)) match {
+        case (SpatialKeyClass, TileClass) => spatialTileRead
+        case (SpatialKeyClass, MultibandTileClass) => spatialMultibandTileRead
+        case _ =>
+          throw new Exception(s"Unable to read single or multiband tiles from file: ${(header.keyClass, header.valueClass)}")
+      }
+    } else {
+      /**
+        * Legacy GeoTrellis Layers compact
+        * TODO: remove in GT 4.0
```
Comment (echeipesh) [inline store/src/main/scala/geotrellis/store/GeoTrellisRasterSource.scala]: Why is this safe to remove in 4.0 ?

File: project/Settings.scala (hunk 1 above)
Comment (pomadchin) [reply project/Settings.scala]: @echeipesh yes, it should be set per project and for some reason setting through `ThisBuild` doesn't work. I tried it in geotrellis-pointcloud & geotrellis-server.

File: store/src/main/scala/geotrellis/store/GeoTrellisRasterSource.scala (hunk 2 above)
Comment (pomadchin) [reply store/src/main/scala/geotrellis/store/GeoTrellisRasterSource.scala]: @echeipesh mb we can drop old layers? :D

**Comment by pomadchin** [comment]
@echeipesh I created a new issue btw https://github.com/geotrellis/geotrellis-server/issues/219

File: store/src/main/scala/geotrellis/store/GeoTrellisRasterSource.scala (hunk 2 above)
Comment (echeipesh) [reply store/src/main/scala/geotrellis/store/GeoTrellisRasterSource.scala]: Seems like an unkind  thing to do if its not buying anybody anything.

File: project/Settings.scala (hunk 1 above)
Comment (echeipesh) [reply project/Settings.scala]: 👍 TIL

File: store/src/main/scala/geotrellis/store/GeoTrellisRasterSource.scala [hunk 3]
```diff
@@ -174,25 +174,43 @@ object GeoTrellisRasterSource {
...
-          .withContext(tiles =>
-            tiles.map{ case(key, tile) => (key, tile.subsetBands(bands)) }
-          )
-      case _ =>
-        throw new Exception(s"Unable to read single or multiband tiles from file: ${(header.keyClass, header.valueClass)}")
+
+    if (!header.keyClass.contains("spark")) {
+      (Class.forName(header.keyClass), Class.forName(header.valueClass)) match {
+        case (SpatialKeyClass, TileClass) => spatialTileRead
+        case (SpatialKeyClass, MultibandTileClass) => spatialMultibandTileRead
```
Comment (echeipesh) [inline store/src/main/scala/geotrellis/store/GeoTrellisRasterSource.scala]: This is mostly style comment that was generated from this line of thinking:

- We get the class class when the class loader is good and match
- If the class is old, we can't get a class class, so we have to check the strings
- And to do that, we have to do a sub-string check
- So wait, we made a class and did a sub-string check to avoid doing a string match?
- Is it slower, I don't know, probably doesn't matter...

TLDR: I think the original formulation is much clearer and potentially saves somebody thinking useless thoughts:

https://github.com/geotrellis/geotrellis-server/blob/89f297c54c12825512716bbcb7c65334a51d4014/core/src/main/scala/geotrellis/store/GeoTrellisRasterSourceLegacy.scala#L237-L244

Disregard this comment if you have a clear preference for doing it this way.

**Comment by pomadchin** [comment]
Replaced by https://github.com/locationtech/geotrellis/pull/3222
